samples/bluetooth/enocean/src/main.c

```diff
@@ -85,6 +85,21 @@ static void enocean_commissioned(struct bt_enocean_device *device)
 	}
 }
 
+static void enocean_decommissioned(struct bt_enocean_device *device)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(&device->addr, addr, sizeof(addr));
+	printk("EnOcean Device decommissioned: %s\n", addr);
+
+	for (int i = 0; i < 4; ++i) {
```

Comment:
Please change iterator to `size_t` type and also please use define for magic number here or put a comment what is 4 here
```suggestion
	for (size_t i = 0; i < 4; ++i) {
```

Reply:
This is the same as what is done in commissioned, so if it is going to be changed here it has to be changed in commissioned too. This is also sample code, and putting a comment saying that it blink the led 4 times is strange and repeat what the code does.
I would also like to keep int, as int is more efficient, and there is no reason for this to be a size_t. So I would like to keep int.

Reply:
Why  `int` would be more efficient that `size_t` here ?

Reply:
I can not see why size_t is better here, as this is just a counter.

Reply:
```
7.17 Common definitions <stddef.h>
size_t which is the unsigned integer type of the result of the sizeof operator
...
7.18.3 Limits of other integer types
— limit of size_t
SIZE_MAX 65535
```
IMHO Since `sizeof` is not used here there is no intention in `size_t`. Even more `int` is aligned with machine word and will be more platform-independent in the general cases. That caused a more optimal compiler backend result.

Reply:
> Even more `int` is aligned with machine word and will be more platform-independent in the general cases. That caused a more optimal compiler backend result.

Both are machine word aligned.

Regarding this:

> 7.18.3 Limits of other integer types
> — limit of size_t
> SIZE_MAX 65535

in the beginning of the paragraph:

> Each instance of these macros shall be replaced by a constant expression suitable for use in #if preprocessing directives, and this expression shall have the same type as would an expression that is an object of the corresponding type converted according to the integer promotions. **Its implementation-defined value shall be equal to or greater in magnitude (absolute value) than the corresponding value given below, with the same sign.**

Not taking any side, but as per what I see using magic numbers as well as int in for-loops is a common practice in Zephyr/NCS and OK to be used in code.

Reply:
> Both are machine word aligned.

Actually no, I'm wrong, it is only size_t that is aligned.

Reply:
> > Even more `int` is aligned with machine word and will be more platform-independent in the general cases. That caused a more optimal compiler backend result.
> 
> Both are machine word aligned.
> 

Nop.

```
https://www.onlinegdb.com/online_c_compiler
```
```
/******************************************************************************

                            Online C Compiler.
                Code, Compile, Run and Debug C program online.
Write your code in this editor and press "Run" button to compile and execute it.

*******************************************************************************/

#include <stdio.h>

int main()
{
    size_t a = 0;
    int b = 0;
    
    printf("size a: %ld, size b: %ld", sizeof(a), sizeof(b));

    return 0;
}
```
```
size a: 8, size b: 4
```

Reply:
My initial comment was about style/consistency with other samples not about performance, it seems for me to be the same for `int` or `size_t` for our architecture but I might be wrong here. I assumed that we use more commonly `size_t` as loop iterator in the NCS than `int` especially in the new code.
About magic number I would say that it is good to not use them particularly in case when the same number occurs few times even if it seems to be a common practice. Anyway this is a sample, I am not going to argue about it, you can decide what you prefer